src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor() {
+    super();
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    Animal.alive.push(this);
+  }
+
+  bite(herb) {
+    if (herb.hidden === false) {
```

Comment:
[CODE PERFORMANCE] - The condition `herb.hidden === false` can be simplified to `!herb.hidden` for better readability and to follow common JavaScript practices.

Reply:
i disagree with you